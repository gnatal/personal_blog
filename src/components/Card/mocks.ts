import type { ICardProps } from '.';

const cardMock:ICardProps = {
  imageSrc: '/images/cat1.jpeg',
  text: 'Cats are so cute that you can\'t resist looking into it',
  imageAlt: 'Cute cat',
  title: 'Cutest cat in the neighborhood',
  tags: [],
};

export default {
  cardMock,
};
