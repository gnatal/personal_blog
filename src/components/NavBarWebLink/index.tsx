import React from 'react';

function NavBarWebLink({
  link,
  placeholder,
}: {
  link: string;
  placeholder: string;
}) {
  return (
    <a
      href={link}
      className='block mt-4 md:mt-2 text-md sm:inline-block lg:mt-0 text-teal-100 hover:text-white mr-4'
    >
      {placeholder}
    </a>
  );
}

export default NavBarWebLink;
