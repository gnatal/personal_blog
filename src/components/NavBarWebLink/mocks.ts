const link1 = 'https://tailwindcss.com/';
const placeholder1 = 'tailwind';

export default {
  link1,
  placeholder1,
};
