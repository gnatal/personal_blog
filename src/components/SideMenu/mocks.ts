const link1 = 'asd';

export default {
  link1,
};
