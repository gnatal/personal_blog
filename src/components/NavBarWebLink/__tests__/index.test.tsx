import React from 'react';
import renderer from 'react-test-renderer';
import NavBarWebLink from '..';
import mocks from '../mocks';

// describe('Testing the nav link in web devices (md or larger)', () => {

// });

test('It should render a link', () => {
  const component = renderer.create(
    <NavBarWebLink link={mocks.link1} placeholder={mocks.placeholder1} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
