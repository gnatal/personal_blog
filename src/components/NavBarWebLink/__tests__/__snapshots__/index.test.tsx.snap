// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render a link 1`] = `
<a
  className="block mt-4 md:mt-2 text-md sm:inline-block lg:mt-0 text-teal-100 hover:text-white mr-4"
  href="https://tailwindcss.com/"
>
  tailwind
</a>
`;
